cmake/BundledLibreSSL.cmake

```diff
@@ -60,6 +60,7 @@ function(use_libre_ssl SOURCE_DIR BINARY_DIR)
     # Set variables
     set(OPENSSL_FOUND "YES" CACHE STRING "" FORCE)
     set(OPENSSL_INCLUDE_DIR "${LIBRESSL_BIN_DIR}/include" CACHE STRING "" FORCE)
+    set(OPENSSL_INCLUDE_DIRS "${OPENSSL_INCLUDE_DIR}" CACHE STRING "" FORCE)  # workaround for libwebsockets
```

Comment:
If some thirdparties require this maybe OPENSSL_INCLUDE_DIRS should be added to FindOpenSSL.cmake as well.

Reply:
I think this is more a bug in `libwebsockets`.  It should be reading `OPENSSL_INCLUDE_DIR`, but (in the case when OpenSSL is part of the project being compiled and not installed on the system) it looks for `OPENSSL_INCLUDE_DIRS` instead.

A better solution may be to patch `libwebsockets` to correctly look for `OPENSSL_INCLUDE_DIR` (and submit the patch upstream).  Do you think it's worth doing that?

Reply:
If you are willing to do that, I think it's a great idea to contribute this improvement to libwebsockets.

Reply:
I'm not sure if it's a bug or not because I saw that in case of other 3rd party libraries both *INCLUDE_DIR and *INCLUDE_DIRS variables are defined being interchangeable. That's why I thought this might be the case here too, so it would worth defining both if any other library would use either of them. If it's actually a bug I think this workaround is okay it's not worth patching it.

Reply:
According to [this](https://discourse.cmake.org/t/findboost-difference-between-boost-include-dir-and-boost-include-dirs/855), INCLUDE_DIR is its own include dir while INCLUDE_DIRS is all of the include dirs needed to use the project, at least in case of Boost. [They are identical](https://github.com/Kitware/CMake/blob/master/Modules/FindBoost.cmake#L2351), though, so I'm not sure if that's actually the case.

Reply:
I think it's very likely to be a bug.  Suggestion: I'll leave this as it is for now and submit a patch to `libwebsockets`.  If they accept it, and a new version with the fix becomes availabe, I will upgrade `libwebsockets` and remove this workaround.

Reply:
I'm okay with that, that should clear all the ambiguity eventually.

Reply:
I have opened this PR: https://github.com/warmcat/libwebsockets/pull/2535.  I haven't got any response yet, but I'll keep an eye on it.